docs/architecture/lb/nginx.md

```diff
@@ -0,0 +1,270 @@
+
+## Install NGINX Open Source
+
+Refer to the NGINX documentation for instructions on installing NGINX **with the stream module**, which is required for TCP/UDP load balancing. For example, on Ubuntu:
+```bash
+sudo apt update
+sudo apt -y install nginx libnginx-mod-stream
+```
+
+## Install NGINX Plus
+
+Refer to the NGINX documentation for instructions on purchasing the license and installation. For example, on Ubuntu:
+```bash
+sudo mkdir -p /etc/ssl/nginx
+
+sudo apt update
+sudo apt-get install apt-transport-https lsb-release ca-certificates wget gnupg2 ubuntu-keyring
+
+# Subscribe to NGINX Plus to obtain nginx-repo.key and nginx-repo.crt
+sudo cp nginx-repo.key nginx-repo.crt /etc/ssl/nginx/
+
+wget -qO - https://cs.nginx.com/static/keys/nginx_signing.key | gpg --dearmor | sudo tee /usr/share/keyrings/nginx-archive-keyring.gpg >/dev/null
+printf "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] https://pkgs.nginx.com/plus/ubuntu `lsb_release -cs` nginx-plus\n" | sudo tee /etc/apt/sources.list.d/nginx-plus.list
+
+sudo wget -P /etc/apt/apt.conf.d https://cs.nginx.com/static/files/90pkgs-nginx
+
+sudo apt-get update
+sudo apt-get install nginx-plus
+```
+
+```bash
+nginx -v
+```
+
+## Fine-tune NGINX
+2. (Optional) Refer to the NGINX documentation for instructions on fine-tuning NGINX performance. For example, you can update the `events` section in your NGINX configuration file:
+
+`/etc/nginx/nginx.conf`
+```conf
+events {
+    worker_connections 20480;
+    multi_accept on;
+    use epoll;
+}
+```
+Please note that actual load balancer fine-tuning is beyond the scope of the SC4S team's responsibility.
+
+## Preserving Source IP
+By default, NGINX overwrites the source IP with the load balancer's IP. While some users accept this behavior, it is recommended to preserve the original source IP of the message.
+
+NGINX provides three methods to preserve the source IP:
+
+| Method                      | Protocol   |
+|-----------------------------|------------|
+| PROXY protocol              | TCP*       |
+| Transparent IP              | TCP/TLS    |
+| Direct Server Return (DSR)  | UDP        |
+
+* TLS PROXY protocol support in SC4S is scheduled for implementation.
+
+Examples for setting up NGINX with the PROXY protocol and DSR are provided below. The Transparent IP method requires complex network configuration. For more details, refer to [this NGINX blog post](https://www.f5.com/company/blog/nginx/ip-transparency-direct-server-return-nginx-plus-transparent-proxy).
+
+
+## Option 1: Configure NGINX with the PROXY Protocol
+
+### Advantages:
+- Easy to set up
+
+### Disadvantages:
+- Available only for TCP, not for UDP or TLS
+- Overwriting the source IP in SC4S is not ideal; the `SOURCEIP` is a hard macro and only `HOST` can be overwritten
+- Overwriting the source IP is available only in SC4S versions greater than 3.4.0
+
```

Comment:
- Available only for TCP, not for UDP or TLS.
- Overwriting the source IP in SC4S is not a best practice; the `SOURCEIP` is a hard macro and only `HOST` can be overwritten.
- Overwriting the source IP is available only in SC4S versions greater than 3.4.0.